src/x/serialize/decoder.go

```diff
@@ -237,3 +237,8 @@ func (d *decoder) Duplicate() ident.TagIterator {
 	}
 	return iter
 }
+
+func (d *decoder) Rewind() {
+	d.checkedData.IncRef()
```

Comment:
Think you want to DecRef after "d.Reset(...)" otherwise this we will never reach NumRef() == 0.